packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -99,19 +301,107 @@ export default function Indicator({ children }) {
     }
   }, [])
 
+  const trackEventProps = { buildId, siteId, orgId }
+
+  if (buildStatus === `BUILDING`) {
+    return (
+      <PreviewIndicator>
+        <GatsbyIndicatorButton
+          tooltipContent={`Building a new preview`}
+          showSpinner={true}
+          overrideShowTooltip={true}
+        />
+        <LinkIndicatorButton
+          tooltipContent={`Copy link`}
+          active={true}
+          {...trackEventProps}
+        />
+        <InfoIndicatorButton {...trackEventProps} />
+      </PreviewIndicator>
+    )
+  }
+
+  if (buildStatus === `ERROR`) {
+    return (
+      <PreviewIndicator>
+        <GatsbyIndicatorButton
+          tooltipContent={
+            <BuildErrorIndicatorTooltip
+              siteId={siteId}
+              orgId={orgId}
+              buildId={currentBuild.id}
+            />
+          }
+          overrideShowTooltip={true}
+          active={true}
+        />
+        <LinkIndicatorButton {...trackEventProps} />
+        <InfoIndicatorButton {...trackEventProps} />
+      </PreviewIndicator>
+    )
+  }
+
+  if (buildId && buildId === buildInfo?.currentBuild?.id) {
+    const updatedDate = formatDistance(
+      Date.now(),
+      new Date(buildInfo?.currentBuild?.createdAt),
+      { includeSeconds: true }
+    )
+    return (
+      <PreviewIndicator>
+        <GatsbyIndicatorButton active={true} />
+        <LinkIndicatorButton
+          tooltipContent={`Copy link`}
+          active={true}
+          {...trackEventProps}
+        />
+        <InfoIndicatorButton
+          tooltipContent={`Preview updated ${updatedDate} ago`}
+          active={true}
+          {...trackEventProps}
+        />
+      </PreviewIndicator>
+    )
+  }
+
+  if (
+    buildId &&
+    buildId !== buildInfo?.latestBuild?.id &&
+    buildStatus === `SUCCESS`
+  ) {
+    return (
+      <PreviewIndicator>
+        <GatsbyIndicatorButton
+          tooltipContent={
+            <BuildSuccessIndicatorTooltip
+              isOnPrettyUrl={buildInfo?.isOnPrettyUrl}
+              sitePrefix={buildInfo?.siteInfo?.sitePrefix}
+              buildId={buildId}
+              siteId={siteId}
+              orgId={orgId}
+            />
+          }
+          overrideShowTooltip={true}
+          active={true}
+          onClick={() =>
+            newPreviewAvailableClick({
+              isOnPrettyUrl: buildInfo?.isOnPrettyUrl,
+              sitePrefix: buildInfo?.siteInfo?.sitePrefix,
+            })
+          }
+        />
+        <LinkIndicatorButton {...trackEventProps} />
+
+        <InfoIndicatorButton {...trackEventProps} />
+      </PreviewIndicator>
+    )
+  }
```

Comment:
Can we maybe create a propBuilder instead, where you build an object with the props and than pass it along the previewindicator? You could even move it to separate functions. I feel that would help reading the outcome of this component.

Reply:
Are you suggesting only to return `PreviewIndicator` and its children once and just pass them the props that we get from the prop builder?

Reply:
That makes it more readable to me yes. Now every if statement makes me think we're rendering a different UI but actually we're changing props. Maybe it's just me :joy:

Reply:
Haha! A prop builder function would dry this up a bit too!

Reply:
@wardpeet are you suggesting to refactor it back to how it was before this PR https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-gatsby-cloud/src/components/GatsbyIndicatorButton.js#L97-L106 ?

Reply:
Fixed in a new commit to render only one path, the conditionals just set different props now and the buttons handle how they should render themselves.